src/Rules.UnitTests/LocalRuleMetadataProviderTests.cs

```diff
@@ -88,12 +88,22 @@ void CheckRule(string fullResourceName)
                 Console.WriteLine("Checking " + fullResourceName);
 
                 SonarCompositeRuleId ruleId = GetCompositeRuleIdFromResourceName(fullResourceName);
-                var expectedDescription = GetEmbeddedRuleDescription(fullResourceName);
+                var expectedRuleInfo = GetEmbeddedRuleDescription(fullResourceName);
 
                 var actual = testSubject.GetRuleInfo(ruleId);
 
                 actual.FullRuleKey.Should().Be(ruleId.ToString());
-                actual.Description.Should().Be(expectedDescription);
+                actual.Description.Should().Be(expectedRuleInfo.Description);
+                
+                if (actual.DescriptionSections.Count > 0)
```

Comment:
Shouldn't this be checking `expectedRuleInfo.DescriptionsSections.Count > 0` instead?

Currently, if the `actual.Descriptions.Count` incorrectly returns no sections when we are expecting some, the tests won't pick it up.

Reply:
I think the problem here is that `actual.DescriptionSections.Should().BeEquivalentTo(expectedRuleInfo.DescriptionSections);` should be outside of the IF